sklearn/utils/_response.py

```diff
@@ -72,15 +73,15 @@ def _get_response_values(
     if is_classifier(estimator):
```

Comment:
do you think this whole block could be simplified / explained? I have a hard time following all these different code paths, and trying to figure out if they actually cover all possible input types, and realized this is kind of a symptom of how this is written.

Reply:
I am not sure where to put the cursor here. Mainly, the issue is that we need a compressed format `(n_samples,)` or `(n_samples, n_outputs)` (i.e. at least our metrics need to). So we leverage `pos_label` to provide the correct array.

Reply:
I created new function and try to better explain what we intend to do.

Reply:
LOVE the new implementation.